lib/pacer.go

```diff
@@ -0,0 +1,170 @@
+package vegeta
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+// A Pacer defines the rate of hits during an Attack by
+// returning the duration an Attacker should wait until hitting
+// the next Target. If the second return value is true, the
+// attack will terminate.
+type Pacer interface {
+	Pace(elapsedTime time.Duration, elapsedHits uint64) (sleep time.Duration, stop bool)
+}
+
+// A PacerFunc is a function adapter type that implements the
+// Pacer interface.
+type PacerFunc func(time.Duration, uint64) (time.Duration, bool)
+
+// Pace implements the Pacer interface.
+func (f PacerFunc) Pace(elapsedTime time.Duration, elapsedHits uint64) (time.Duration, bool) {
+	return f(elapsedTime, elapsedHits)
+}
+
+// Rate sends a constant rate of hits to the target.
+type Rate struct {
```

Comment:
How about naming it `FixedRate` or `FixedRatePacer` in light of the latest changes ?

Also, consider renaming `SinePacer` so that both implementations look similar (that is either use `Pacer` suffix for all of them or none of them)

Reply:
Rate needed to remain named Rate for backwards compatibility. We could use a [type alias](https://go.googlesource.com/proposal/+/master/design/18130-type-alias.md), I suppose, if we're willing to restrict users of the library to go1.9+. WDYT?

Reply:
Oops! Forgot about backward-compatibility. 

Using type alias looks reasonable to me. That's for @tsenart to decide :)

Reply:
I never used type aliases before. At first glance, it seems fine. I don't mind restricting this library to go1.9+. Are there any other caveats?

As for the name, what do you think of `ConstantPacer` or `SteadyPacer`?

Reply:
I think `SteadyPacer` might not be exact enough :) `ConstantPacer` looks great!